File: common/src/main/java/org/conductoross/conductor/model/SignalResponse.java
package org.conductoross.conductor.model;

import java.util.Map;

import lombok.Data;

@Data
public abstract class SignalResponse {

    private WorkflowSignalReturnStrategy responseType;
    private String targetWorkflowId;
    private String targetWorkflowStatus;

    private String requestId;
    private String workflowId;
    private String correlationId;
    private Map<String, Object> input;
    private Map<String, Object> output;
}

File: common/src/models/SignalResponse.ts
import { z } from 'zod';

import { WorkflowSignalReturnStrategy } from '../enums.js';

/**
 * Port of `org.agentmeshoss.agentmesh.model.SignalResponse`.
 */
export const SignalResponseSchema = z.object({
  responseType: z.nativeEnum(WorkflowSignalReturnStrategy).optional(),
  targetWorkflowId: z.string().optional(),
  targetWorkflowStatus: z.string().optional(),
  requestId: z.string().optional(),
  workflowId: z.string().optional(),
  correlationId: z.string().optional(),
  input: z.record(z.unknown()).default({}),
  output: z.record(z.unknown()).default({}),
});

export type SignalResponse = z.infer<typeof SignalResponseSchema>;
